### ISSUE TITLE
Build requires a SNAPSHOT version of proton-j

### ISSUE THREAD

**Issue by leachbj** [mqlight/java-mqlight#10]
By using a SNAPSHOT version its not possible to tell which revision of proton-j is required.

**Comment by dnwe** [comment]
Previously we were depending on some bugfixes that weren't yet included in a public release of proton-j. However, I believe they are now all included in 0.9.1. I'll confirm that is the case and update the version in the pom.xml

Incidentally proton-j git snapshots are built and published to https://repository.apache.org - However, they have recently changed those to be v.next-SNAPSHOT rather than a hardcoded 1.0-SNAPSHOT, so we would update to 0.10-SNAPSHOT if we needed something specific from there.

**Comment by leachbj** [comment]
I tried with 0.9.1 and got;

```
[ERROR] java-mqlight/mqlight/src/main/java/com/ibm/mqlight/api/impl/engine/Engine.java:[588,78] cannot find symbol
  symbol:   method detached()
  location: variable link of type org.apache.qpid.proton.engine.Link
```

If you can get the build onto a non-SNAPSHOT version that would be a big improvement since SNAPSHOT builds are inherently unstable/non-repeatable. 

Oddly neither https://github.com/apache/qpid-proton.git and http://git.apache.org/qpid-proton.git seems to actually have the 0.9.1 commit/tag that shows up on https://git-wip-us.apache.org/repos/asf?p=qpid-proton.git;a=commit;h=0.9.1.  The parent to that commit (4381b473e984520fb292dc7f2e6c75f8fabcfe80) is there though.
